connexion/validators/form_data.py

```diff
@@ -94,30 +81,29 @@ def validate(self, data: FormData) -> None:
                 if isinstance(v, str):
                     form_data[k] = data.getlist(k)
                 elif isinstance(v, UploadFile):
-                    file_data[k] = data.getlist(k)
+                    # Replace files with empty strings for validation
+                    file_data[k] = ""
 
             data = self.uri_parser.resolve_form(form_data)
             # Add the files again
             data.update(file_data)
         else:
             data = {k: data.getlist(k) for k in data}
 
-        for k, param_defn in props.items():
-            if k in data:
-                if param_defn.get("format", "") == "binary":
-                    # Replace files with empty strings for validation
-                    data[k] = ""
-                    continue
+        return data
 
-                try:
-                    data[k] = coerce_type(param_defn, data[k], "requestBody", k)
-                except TypeValidationError as e:
-                    logger.exception(e)
-                    errs += [str(e)]
+    def _validate_strictly(self, data: FormData) -> None:
+        form_params = data.keys()
+        spec_params = self.schema.get("properties", {}).keys()
+        errors = set(form_params).difference(set(spec_params))
```

Comment:
Is the set conversion actually needed here? Doesn't the keyview support set operations directly?

Reply:
Only via bitwise operations. So `form_params - set(spec_params)` would work and I guess that's easier to understand. I'll do it in a separate PR though, since I'm afraid the pipeline on this PR would start failing due to [an issue](https://github.com/PyCQA/isort/issues/2083) for which I already included a fix in #1628.